### ISSUE TITLE
Errors highlighting color

### ISSUE THREAD

**Issue by rbaumi** [microsoft/TypeScript-Sublime-Plugin#550]
Is it possible to change the color (or maybe the style) of errors highlighting? What I mean is the color of the `Keywords` from the theme.

![](https://s16.postimg.org/9jywfs5ud/Screenshot_from_2016_10_25_22_46_01.png)

Any way to change the color and / or mark it on the line number (like linter does)?

**Comment by rbaumi** [comment]
Ok, I could manage to do it by changing lines 164 in file [typescript/listeners/idle.py](https://github.com/Microsoft/TypeScript-Sublime-Plugin/blob/master/typescript/listeners/idle.py) to:

```py
view.add_regions(region_key, error_regions, "invalid.illegal", "",
    sublime.DRAW_SOLID_UNDERLINE)
```

It would be good if both color and mark style was an config option. Also would be nice to be able to allow for marking the whole line (`start_offset` variable).

**Comment by zhengbli** [comment]
Thanks for nivestigating! You are always welcome to send a PR :)

**Comment by alexburner** [comment]
Thanks for the patch @rbaumi, I can see errors now!

**Comment by rgant** [comment]
A work around for this is to add this to your .tmTheme in Sublime:

```xml
<dict>
  <key>name</key>
  <string>TypeScript Errors</string>
  <key>scope</key>
  <string>keyword</string>
  <key>settings</key>
  <dict>
    <key>foreground</key>
    <string>#F92672</string>
  </dict>
</dict>
<dict>
  <key>name</key>
  <string>Keyword</string>
  <key>scope</key>
  <string>keyword.control, keyword.operator, keyword.other</string>
  <key>settings</key>
  <dict>
    <key>foreground</key>
    <string>#cc99cc</string>
  </dict>
</dict>
```

**Comment by betorobson** [comment]
@rgant these change on .tmTheme should display erros with gutter and outline?
I have tried it but I got any change :-(

**Comment by rgant** [comment]
I don't use GitGutter (weird I know) so I'm not 100% what you are talking about. My styles above only apply to the Sublime view of the file.

<img width="362" alt="screen shot 2018-11-08 at 10 14 17" src="https://user-images.githubusercontent.com/710553/48207685-28196d80-e33f-11e8-97f9-70b4a1fd7ae7.png">

The pink squiggles in the screen shot above are scope "keyword" and are from TypeScript's error checking. The purple "import" and "from" are scope "keyword.control" (or "keyword.operator" or "keyword.other").

**Comment by betorobson** [comment]
I am looking for a way to display Typescript Erros with gutters and outline like this:
![screenshot from 2018-11-08 13-29-44](https://user-images.githubusercontent.com/13212208/48208675-7d16ad00-e35a-11e8-8ee1-6f40cb949f7d.png)

**Comment by rgant** [comment]
In order do do that you would need to change the `icon` parameter to the [add_regions code for displaying the errors](/Microsoft/TypeScript-Sublime-Plugin/blob/master/typescript/listeners/idle.py#L161). See #681 where I am changing the `scope` parameter. `add_regions` documented here: https://www.sublimetext.com/docs/3/api_reference.html#sublime.View

**Comment by rgant** [comment]
I believe this is resolved by merging #681 and this issue can be closed.

**Comment by orta** [comment]
👍
